src/index.jsx

```diff
@@ -25,21 +27,21 @@ registerIcons();
 subscribe(APP_READY, () => {
   ReactDOM.render(
     <AppProvider store={configureStore()}>
-      <LoggedInRedirect>
-        <HeaderLayout>
-          <Switch>
-            <Route exact path="/">
-              <Redirect to={LOGIN_PAGE} />
-            </Route>
-            <Route exact path={LOGIN_PAGE} component={LoginPage} />
-            <Route exact path={REGISTER_PAGE} component={RegistrationPage} />
-            <Route exact path={RESET_PAGE} component={ForgotPasswordPage} />
-            <Route exact path="/password_reset_confirm/:token/" component={ResetPasswordPage} />
-            <Route path="/notfound" component={NotFoundPage} />
-            <Route path="*" component={NotFoundPage} />
-          </Switch>
-        </HeaderLayout>
-      </LoggedInRedirect>
+      <HeaderLayout>
+        <Switch>
+          <Route exact path="/">
+            <Redirect to={PAGE_NOT_FOUND} />
+          </Route>
+          <UnAuthOnlyRoute exact path={LOGIN_PAGE} component={LoginPage} />
+          <UnAuthOnlyRoute exact path={REGISTER_PAGE} component={RegistrationPage} />
+          <UnAuthOnlyRoute exact path={RESET_PAGE} component={ForgotPasswordPage} />
+          <Route exact path="/password_reset_confirm/:token/" component={ResetPasswordPage} />
```

Comment:
nit: can we move this path to constants as well?

Reply:
sure.